src/main/java/com/mackervoy/calum/mud/content/ContentController.java

```diff
@@ -1,59 +1,69 @@
+
+import java.io.ByteArrayOutputStream;
+import java.util.Optional;
 
 import javax.ws.rs.GET;
 import javax.ws.rs.Path;
 import javax.ws.rs.Produces;
 import javax.ws.rs.QueryParam;
-import javax.ws.rs.core.MediaType;
-
 import org.apache.jena.rdf.model.Model;
 import org.apache.jena.rdf.model.ModelFactory;
-import org.apache.jena.rdf.model.Property;
 import org.apache.jena.rdf.model.Resource;
+import org.apache.jena.rdf.model.ResourceFactory;
 import org.apache.jena.vocabulary.RDF;
 
-import com.mackervoy.calum.mud.MUDApplication;
-
 /**
  * @author Calum Mackervoy
- * Provides endpoints for client-side applications to request content 
+ * Provides endpoints for client-side applications to request content
  * (e.g. "please describe this object")
  */
 @Path("/content/")
 public class ContentController {
-	@GET
-    @Produces(MediaType.TEXT_PLAIN)
-    public String getContent(@QueryParam("uri") String uri) {
-		// bounds checking - could provide a catch-all describer this way
-		if(uri == null) uri = "";
-		
+  @GET
+  @Produces("text/turtle")
+  public String getContent(@QueryParam("uri") String uri) {
+		final String safeUri = uri == null ? "" : uri;
+		final Model m = ModelFactory.createDefaultModel();
+    m.read(safeUri);
+
+		return getDescriber(safeUri, m)
+			.map(describer -> describer.describe(m, safeUri))
+      .map(description -> descriptionToSensesModel(description))
+      .map(senseModel -> modelToTurtle(senseModel))
+			.orElse(null);
+  }
+	
+	private Optional<IContentDescriber> getDescriber(String uri, Model m) {
 		// check if I have a describer for the URI
-		DescriberFactory factory = new DescriberFactory();
-		IContentDescriber describer = factory.getDescriber(uri);
-		Model m = ModelFactory.createDefaultModel();
-    	m.read(uri);
+    DescriberFactory factory = new DescriberFactory();
+    IContentDescriber describer = factory.getDescriber(uri);
 		
 		// if I don't, try the RDF type
-		if(describer == null) {
-			Resource r = m.getResource(uri);
-	    	String type = r.getPropertyResourceValue(RDF.type).toString();
-	    	System.out.println("Retrieved type: " + type);
-	    	describer = factory.getDescriber(type);
-		}
-		
-		// found a Describer, return 200 with content
-		if(describer != null) {
-			return describer.describe(m, uri);
-		}
-		
-		// could not find a describer - return 204
-    	return null;
+		if (describer == null) {
+      Resource r = m.getResource(uri);
+      String type = r.getPropertyResourceValue(RDF.type).toString();
+      describer = factory.getDescriber(type);
     }
+
+		return Optional.ofNullable(describer);
+  }
+
+  private Model descriptionToSensesModel(String visualDescription) {
+    Model m = ModelFactory.createDefaultModel();
+    Resource senses = ResourceFactory.createResource("http://localhost:8080/mud/content/#senses"); // not sure what this ought to be
+    m.add(senses, MUD.sight, visualDescription);
+    return m;
+  }
+
+  private String modelToTurtle(Model m) {
```

Comment:
Good shout - we should move this to a universal utility function at some point or some sort of abstract class for Controllers to extend